firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateAppClient.java

```diff
@@ -79,10 +109,185 @@ private void redirectToPlayForAabUpdate(String downloadUrl, Activity currentActi
     this.updateTask.updateProgress(updateState);
   }
 
```

Comment:
Since updateApk logic is pretty complex, it might be worth refactoring all of that in an UpdateApkClient